### PR TITLE
docs(cdk/layout): better explain how layout breakpoints work

### LINKED ISSUE #11788: Better explaination of Breakpoints
#### Bug, feature request, or proposal:
Proposal

#### What is the expected behavior?
A clear explanation of the built-in breakpoints

#### What is the current behavior?
Right now we just have a link to https://material.io/guidelines/layout/responsive-ui.html#responsive-ui-breakpoints
(and that redirects to https://material.io/design/layout/responsive-layout-grid.html#columns-gutters-margins)

and we see the available values but no explanation of what they mean :)

#### What are the steps to reproduce?
Look at docs at https://material.angular.io/cdk/layout/overview

### PR DESCRIPTION
**Pull request by jelbourn**
I'm copying the actual breakpoints here for reference because it's much
easier to parse than looking at the archived Material Design site.

I'm also working on a follow-up to add an overview example for
`BreakpointObserver`.

Fixes #11788

### CODE REVIEW

File: src/cdk/layout/layout.md [hunk 1]
```diff
@@ -2,18 +2,24 @@ The `layout` package provides utilities to build responsive UIs that react to sc
 
 ### BreakpointObserver
 
-`BreakpointObserver`  is a utility for evaluating media queries and reacting to changes in the results of those queries.
+A layout **breakpoint** is viewport size threshold at which a layout shift can occur. The viewport
+size ranges between breakpoints correspond to different standard screen sizes. 
+
+`BreakpointObserver` lets you evaluate media queries to determine the current screen size and
+react to changes when the viewport size changes and crosses a breakpoint.
```
Comment (crisbeto) [inline src/cdk/layout/layout.md:9]: Should we mention that the `BreakpointObserver` isn't limited to just screen size viewports? You can use it for things like detecting dark mode or reduced motion.

File: src/cdk/layout/layout.md [hunk 2]
```diff
@@ -2,18 +2,24 @@ The `layout` package provides utilities to build responsive UIs that react to sc
...
-The `isMatched` method is used to evaluate one or more media queries against the current viewport
-size.
 ```ts
 const isSmallScreen = breakpointObserver.isMatched('(max-width: 599px)');
 ```
 
 #### React to changes to the viewport
-The `observe` method is used to get an observable stream that will emit whenever one of the given
-media queries would have a different result.
+You can use the `observe` method to get an observable stream emits whenever the viewport size
```
Comment (crisbeto) [inline src/cdk/layout/layout.md:20]: observable stream emits -> observable stream that emits

File: src/cdk/layout/layout.md [hunk 3]
```diff
@@ -2,18 +2,24 @@ The `layout` package provides utilities to build responsive UIs that react to sc
...
+changes and crosses a breakpoint.
```
Comment (crisbeto) [inline src/cdk/layout/layout.md:21]: Maybe this should say just "crosses a breakpoint"? It isn't going to fire on any viewport resize.

File: src/cdk/layout/layout.md (hunk 1 above)
Comment (jelbourn) [reply src/cdk/layout/layout.md:9]: My thinking is that those things are unlikely to change in the middle of the app running, so you could just use `MediaMatcher` directly.

File: src/cdk/layout/layout.md (hunk 2 above)
Comment (jelbourn) [reply src/cdk/layout/layout.md:20]: Done

File: src/cdk/layout/layout.md (hunk 3 above)
Comment (jelbourn) [reply src/cdk/layout/layout.md:21]: Done

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**